agent/consul/server.go

```diff
@@ -236,6 +242,10 @@ type Server struct {
 	// for the KV tombstones
 	tombstoneGC *state.TombstoneGC
 
+	// endpointMap retains a reference to all registered rpc endpoint handlers
+	// for use in tests where the RPC interface is insufficient. DO NOT USE outside of tests.
+	endpointMap map[string]interface{}
```

Comment:
This lets you build a `Server` and then from test code extract a pointer to the preregistered endpoint handler (like `*ACL`) rather than what the tests used to do which was build a new `ACL` instance for the test (meaning that any RPCs kicked off from other test code would pass through the Server's ACL instance rather than the instance under test.